lib/client.ml

```diff
@@ -49,7 +47,7 @@ let recv_next client =
   (* Now that we have read the header and the message size, we can read the message *)
   let%lwt _, addr = recvfrom !client.socket msg_buffer 0 msg_size [] in
   Lwt_mutex.unlock !client.recv_mutex;
-  Lwt.return (msg_buffer, Peer.from_sockaddr addr)
+  Lwt.return (msg_buffer, peer_from (Address.from_sockaddr addr))
```

Comment:
This is following your comment to switch back to `Peer.t` instead of `Address.t` everywhere.

Reply:
Nice catch! Done.